### PR TITLE
search: Refactor search stack UI widget

### PR DESCRIPTION
**Pull request by fkling**
This makes some updates to the look and feel of the widget. In
particular it changes the closed state of the widget from a small button
to showing the top item. Clicking on the item will open the widget.

The buttons for adding a new entry will now always be shown, even if the
entry was just added.

Annotations can now be added. Those are rendered as markdown blocks
*before* the references block.

More thought has to be put into showing buttons for adding new entries in the closed state.

https://user-images.githubusercontent.com/179026/154087643-7c7337d9-4450-41be-87c3-205f2408f5fc.mp4

## Test plan

- Go to /search or any a file view. The widget should show up, showing buttons to add new entries.
- Click on the button to add an entry.
- Click on the entry to open the widget.
- Click on the "annotations button" to toggle the annotations input.
- Click on the notebook button. The annotations should be converted to markdown blocks before the search/file block.

### CODE REVIEW

File: client/web/src/search/SearchStack.test.tsx [hunk 1]
```diff
@@ -73,7 +84,10 @@ describe('Search Stack', () => {
         })
 
         it('opens and closes', () => {
-            renderSearchStack()
+            const result = renderSearchStack()
+
+            expect(result.asFragment()).toMatchSnapshot()
```
Comment (limitedmage) [inline client/web/src/search/SearchStack.test.tsx:89]: Is a snapshot test valuable here? Maybe we can make this a storybook test instead.

File: client/web/src/stores/searchStack.ts [hunk 2]
```diff
@@ -108,7 +110,7 @@ export function useSearchStack(newEntry: SearchStackEntryInput | null): void {
             // We have to "remove" the entry if the component unmounts.
             return () => {
                 const currentState = useSearchStackState.getState()
-                if (currentState.addableEntry === newEntry) {
+                if (currentState.addableEntry === entry) {
```
Comment (limitedmage) [inline client/web/src/stores/searchStack.ts:113]: Does this equality check actually work? `entry` is likely a new object so it wouldn't match?

**Review by limitedmage — APPROVED** [review]
Looking awesome!!

File: client/web/src/stores/searchStack.ts (hunk 2 above)
Comment (fkling) [reply client/web/src/stores/searchStack.ts:113]: Ugh, good point... let me test that again :D

Comment (fkling) [reply client/web/src/stores/searchStack.ts:113]: It works because the callsites use `useMemo`. I'll add a comment that this is expected.

Comment (fkling) [reply client/web/src/stores/searchStack.ts:113]: Oh, I already have a comment :D

**Pull request merged**